src/compute-client/src/controller/instance.rs

```diff
@@ -878,8 +885,9 @@ where
                     Some(cmd) => cmd(&mut self),
                     None => break,
                 },
-                (replica_id, response) = Self::recv(&mut self.replicas) => {
-                    self.handle_response(response, replica_id);
+                response = self.replica_rx.recv() => match response {
+                    Some((replica_id, response)) => self.handle_response(response, replica_id),
+                    None => break,
```

Comment:
I think this `break` should be a `panic`/`unreachable`. The `Instance` is holding onto the `replica_tx`, so there should always be at least one sender connected.